src/Chat/LmStudioChat.php

```diff
@@ -0,0 +1,316 @@
+            $contents = $response->getBody()->getContents();
+            $this->logger->debug($contents);
+            $json = Utility::decodeJson($contents);
+
+            if (!isset($json['choices']) || empty($json['choices'])) {
+                error_log('❌ LM Studio response missing choices array');
+                error_log('   Response: ' . $contents);
+                throw new \Exception('Invalid LM Studio response: no choices returned');
+            }
+
+            $message = $json['choices'][0]['message'] ?? ['content' => ''];
+
+            if (array_key_exists('tool_calls', $message) && !empty($message['tool_calls'])) {
+                $toolCall = $message['tool_calls'][0];
+                $functionName = $toolCall['function']['name'];
+                $arguments = Utility::decodeJson($toolCall['function']['arguments']);
+
+                return [
+                    'function_name' => $functionName,
+                    'arguments' => $arguments,
+                ];
+            }
+
+            return $message['content'] ?? '';
+        }
+
+        // =================================================================================================================
+        // STREAMING METHODS (Basic stub implementation to satisfy the interface)
+        // =================================================================================================================
+
+        public function generateStreamOfText(string $prompt): StreamInterface
+        {
+            $result = $this->generateText($prompt);
+            return Utils::streamFor($result);
+        }
+
+        /**
+         * @param Message[] $messages
+         */
+        public function generateStreamOfChat(array $messages): StreamInterface
+        {
+            $result = $this->generateChat($messages);
+            return Utils::streamFor($result);
+        }
+
+        /**
+         * @param Message[] $messages
+         */
+        public function generateChatStream(array $messages): StreamInterface
+        {
+            $result = $this->generateChat($messages);
+            return Utils::streamFor($result);
+        }
+
+        /**
+         * @param Message[] $messages
+         */
+        public function generateStreamOfChatOrReturnFunctionCalled(array $messages): array|StreamInterface
+        {
+            $result = $this->generateChatOrReturnFunctionCalled($messages);
+
+            if (is_array($result)) {
+                return $result;
+            }
+
+            return Utils::streamFor($result);
+        }
+
+        // =================================================================================================================
+        // EMBEDDING METHOD
+        // =================================================================================================================
+
+        public function getEmbedding(string $text, string $model = 'default-embed'): array
```

Comment:
Embeddings should be handled by a class implemnenting `EmbeddingGeneratorInterface`